hug/interface.py

```diff
@@ -417,8 +418,7 @@ def __call__(self, *args, **kwargs):
                 self.api.delete_context(context, errors=errors)
                 return outputs(errors) if outputs else errors
 
-        if getattr(self, "map_params", None):
```

Comment:
`getattr` is bit of a smell. While it is sometimes totally fine, here the whole condition can be removed by ensuring `self.map` is always of the same type (dict).